middleware/cors.go

```diff
@@ -102,6 +102,17 @@ func CORSWithConfig(config CORSConfig) echo.MiddlewareFunc {
 			origin := req.Header.Get(echo.HeaderOrigin)
 			allowOrigin := ""
 
+			preflight := req.Method == http.MethodOptions
```

Comment:
I don't think it is needed. Because OPTIONS is used for preflight requests whereas `AllowedMethods` defines allowed methods for simple requests.

Note that other frameworks I tested (play and gin) do not provide send any `Access-Control-Allow-Methods` headers by default.